emulated-integration-tests/src/chains/asset_hub_rococo/mod.rs

```diff
@@ -0,0 +1,44 @@
+pub mod genesis;
+
+// Substrate
+use frame_support::traits::OnInitialize;
+
+// Cumulus
+use crate::chains::rococo::Rococo;
+use emulated_integration_tests_common::{
+	impl_accounts_helpers_for_parachain, impl_assert_events_helpers_for_parachain,
+	impl_assets_helpers_for_parachain, impl_foreign_assets_helpers_for_parachain,
+	impl_xcm_helpers_for_parachain, impls::Parachain, xcm_emulator::decl_test_parachains,
+};
+
+// AssetHubRococo Parachain declaration
+decl_test_parachains! {
+	pub struct AssetHubRococo {
+		genesis = genesis::genesis(),
+		on_init = {
+			asset_hub_rococo_runtime::AuraExt::on_initialize(1);
+		},
+		runtime = asset_hub_rococo_runtime,
+		core = {
+			XcmpMessageHandler: asset_hub_rococo_runtime::XcmpQueue,
+			LocationToAccountId: asset_hub_rococo_runtime::xcm_config::LocationToAccountId,
+			ParachainInfo: asset_hub_rococo_runtime::ParachainInfo,
+			MessageOrigin: cumulus_primitives_core::AggregateMessageOrigin,
+		},
+		pallets = {
+			PolkadotXcm: asset_hub_rococo_runtime::PolkadotXcm,
```

Comment:
Been a while since ive looked at emulator, but arent these redundant? I thought this method was a way to add additional pallets into a runtime rather than reconfigure them? May be wrong tho...

Reply:
Or maybe it was a way to easily get a handle on a pallet from a test.

Reply:
So what happens is that the trait AssetHubRococoParaPallet is defined like:
```
pub trait AssetHubRococoParaPallet {
    type PolkadotXcm;
    type Assets;
    type ForeignAssets;
    type PoolAssets;
    type AssetConversion;
    type Balances;
}
```
And implemented for the emulated chain:
```
impl<N: ::xcm_emulator::Network> AssetHubRococoParaPallet for AssetHubRococo<N> {
    type PolkadotXcm = asset_hub_rococo_runtime::PolkadotXcm;
    type Assets = asset_hub_rococo_runtime::Assets;
    type ForeignAssets = asset_hub_rococo_runtime::ForeignAssets;
    type PoolAssets = asset_hub_rococo_runtime::PoolAssets;
    type AssetConversion = asset_hub_rococo_runtime::AssetConversion;
    type Balances = asset_hub_rococo_runtime::Balances;
}
```

Which is then used to call a runtime function or read an event etc.